app/src/main/java/com/google/cloud/solutions/flexenv/PlayActivity.java

```diff
@@ -333,16 +336,16 @@ public void onConnectionFailed(@NonNull ConnectionResult connectionResult) {}
      * Request that a Servlet instance be assigned.
      */
     private void requestLogger() {
-        firebase.child(IBX + "/" + inbox).removeValue();
-        firebase.child(IBX + "/" + inbox).addValueEventListener(new ValueEventListener() {
+        databaseReference = FirebaseDatabase.getInstance().getReference();
+        databaseReference.child(IBX + "/" + inbox).removeValue();
+        databaseReference.child(IBX + "/" + inbox).addValueEventListener(new ValueEventListener() {
             public void onDataChange(@NonNull DataSnapshot snapshot) {
-                if (snapshot.exists()) {
-                    if(snapshot.getValue() != null) {
-                        fbLog = new FirebaseLogger(firebase, IBX + "/" + snapshot.getValue().toString()
-                                + "/logs");
-                        firebase.child(IBX + "/" + inbox).removeEventListener(this);
-                        fbLog.log(inbox, "Signed in");
```

Comment:
This call to the `log()` method is not required here anymore. It should be moved to the `onActivityResult()` method after the user signs in (approximately in line 225).